File: topic_4/Haskell/15-swap.hs
-- Troca dois elementos de uma lista passando a lista e as posições
-- a serem trocadas

swap [] a b = []
swap [x] a b = [x]
swap xs a b = comeco ++ elemB ++ meio ++ elemA ++ fim
    where
        comeco = take a xs
        elemB = take 1 (drop b xs)
        meio = drop (a+1) (take b xs)
        elemA = take 1 (drop a xs)
        fim = drop (b+1) xs
    
main = do
    a <- readLn :: IO [Int]
    b <- readLn :: IO Int
    c <- readLn :: IO Int
    print $ swap a b c

File: topic_4/Python/15-swap.py
# Troca dois elementos de uma lista passando a lista e as posições
# a serem trocadas

def swap(xs, a, b):
    if xs != [] and len(xs) > 1: # verifica se a lista é vazia ou se só tem um elemento
        xs[a], xs[b] = xs[b], xs[a]
    return xs

print(swap ([], 0, 5) == []) #True
print(swap ([1], 0, 3) == [1]) #True
print(swap ([1,3,4], 1, 2) == [1,4,3]) #True
print(swap ([1,2,3,4,5,6], 2, 5) == [1,2,6,4,5,3]) #True
print(swap ([5,6,7,8,9], 0, 3) == [8,6,7,5,9]) #True